File: test/benchmark/r7rs/python/fib.py
#!/usr/bin/env python3
# R7RS Benchmark: fib (Python)
# Naive recursive Fibonacci - fib(27) = 196418
import time


def fib(n):
    if n < 2:
        return n
    return fib(n - 1) + fib(n - 2)


def main():
    import sys
    sys.setrecursionlimit(10000)
    t0 = time.perf_counter_ns()
    result = fib(27)
    t1 = time.perf_counter_ns()

    if result == 196418:
        print("fib: PASS")
    else:
        print(f"fib: FAIL result={result}")
    print(f"__TIMING__:{(t1 - t0) / 1e6:.3f}")


main()

File: test/benchmark/r7rs/c2mir/fib.c
/* Native C2MIR port of r7rs/fib2.ls. */
extern int printf(const char *, ...);

static int fib(int n) {
    if (n < 2) return n;
    return fib(n - 1) + fib(n - 2);
}

int main(void) {
    int result = fib(27);
    printf(result == 196418 ? "fib: PASS\n" : "fib: FAIL result=%d\n", result);
    return result != 196418;
}
